robottelo/host_helpers/api_factory.py

```diff
@@ -711,6 +711,251 @@ def wait_for_errata_applicability_task(
+            entities = {k: v.read() for k, v in entities.items()}
+
+        # promote to non-Library env if not already present:
+        # skip for 'Library' env selected or passed arg,
+        # any published version(s) will already be in Library.
+        if all(
+            [
+                environment != 'Library',
+                entities['LifecycleEnvironment'].name != 'Library',
+                entities['LifecycleEnvironment'] not in entities['ContentView'].environment,
+            ]
+        ):
+            # promote newest version by id
+            entities['ContentView'].version.sort(key=lambda version: version.id)
+            entities['ContentView'].version[-1].promote(
+                data={'environment_ids': entities['LifecycleEnvironment'].id}
+            )
+            # updated entities after promoting
+            entities = {k: v.read() for k, v in entities.items()}
+
+        if (  # assign env to ak if not present
+            entities['ActivationKey'].environment is None
+            or entities['ActivationKey'].environment.id != entities['LifecycleEnvironment'].id
+        ):
+            entities['ActivationKey'].environment = entities['LifecycleEnvironment']
+            entities['ActivationKey'].update(['environment'])
+            entities = {k: v.read() for k, v in entities.items()}
+        if (  # assign cv to ak if not present
+            entities['ActivationKey'].content_view is None
+            or entities['ActivationKey'].content_view.id != entities['ContentView'].id
+        ):
+            entities['ActivationKey'].content_view = entities['ContentView']
+            entities['ActivationKey'].update(['content_view'])
+
+        # register with now setup entities, using ak
+        entities = {k: v.read() for k, v in entities.items()}
+        result = client.register(
+            activation_keys=entities['ActivationKey'].name,
+            target=self._satellite,
+            org=entities['Organization'],
+            loc=None,
+            force=force,
+        )
+        if result.status != 0:
+            method_error['message'] = (
+                f'Failed to register the host: {client.hostname}.\n{result.stderr}'
+            )
+            return method_error
+        if not client.subscribed:
+            method_error['client'] = client
+            method_error['message'] = (
+                f'Failed to subscribe the host: {client.hostname},'
+                f' to content-view: {entities["ContentView"].name}'
+            )
+            return method_error
+        if rex_key:
+            client.add_rex_key(self._satellite)
+
+        # enable all repositories available to client, using subscription manager,
+        # ie: any repos added to content-view prior to calling this method.
+        # note: this will fail if no repos are available to client from CV/AK
+        if enable_repos:
+            output = client.execute(r'subscription-manager repos --enable \*')
```

Comment:
I know we use this at several places already. I just wonder if it wouldn't be better to do content-override at the AK entity (based on the `enable_repos` param) prior to the client registration, so the repos would get enabled through the AK already.